open-company/open-company-web
# Fix sentry OC-BETA-WEB-D2

Error: https://sentry.io/organizations/opencompany/issues/1089723740/?project=58003&referrer=slack

Bug: if you click New post immediately when the app appear the selected section disappear from Cmail and it cause an error when saving.

The issue is caused by the function that tries to reopen cmail: when you click the New post button we save a cookie to make sure we remember to reopen it if the window is closed. The check to reopen cmail is done with a little delay so the cookie is read and the cmail state is reset.
It's a race condition btw the cmail handling function (save and reopen).

To test:
- open on mainline
- click New post as soon as the button is visible
- wait a few seconds
- [x] the section selected for the new post should disappear and any edit will cause an error (for missing section)
- now switch to this branch
- repeat
- [x] section still there? Good
- repeat a few times to make sure
